tools/rpkg/R/dbQuoteIdentifier__duckdb_connection.R

```diff
@@ -0,0 +1,33 @@
+dbQuoteIdentifier__duckdb_connection <- function(conn, x, ...) {
+  if (is(x, "SQL")) {
+    return(x)
+  }
+  if (is(x, "Id")) {
+    return(SQL(paste0(dbQuoteIdentifier(conn, x@name), collapse = ".")))
+  }
+
+  x <- enc2utf8(x)
+  needs_escape <- !grepl("^[a-zA-Z0-9_]+$", x) | tolower(x) %in% reserved_words(conn)
+  x[needs_escape] <- paste0('"', gsub('"', '""', x[needs_escape]), '"')
+
+  SQL(x, names = names(x))
+}
+
+#' @rdname duckdb_connection-class
+#' @export
+setMethod("dbQuoteIdentifier", signature("duckdb_connection"), dbQuoteIdentifier__duckdb_connection)
+
+setMethod("dbQuoteIdentifier", signature("duckdb_connection", "character"), dbQuoteIdentifier__duckdb_connection)
+setMethod("dbQuoteIdentifier", signature("duckdb_connection", "SQL"), dbQuoteIdentifier__duckdb_connection)
+setMethod("dbQuoteIdentifier", signature("duckdb_connection", "Id"), dbQuoteIdentifier__duckdb_connection)
+
+reserved_words <- function(con) {
```

Comment:
This caches the list of reserved words on first run; let me know if you'd prefer a different strategy.

Reply:
Oh hmmm, of course, this doesn't work with the `translate_sql()` tests which pass in a dummy connection. Maybe it's better to do this `.onLoad`?